File: with-kafka/types/Transaction.ts
export class Transaction {
  transactionId: string;
  senderAccountId: string;
  receiverAccountId: string;
  amount: number;
  currency: string;
  timestamp: Date;

  constructor(senderId: string, recieverId: string, amount: number) {
    this.transactionId = this.generateTransactionId();
    this.senderAccountId = senderId;
    this.receiverAccountId = recieverId;
    this.amount = amount;
    this.currency = "USD";
    this.timestamp = new Date();
  }

  private generateTransactionId(): string {
    return Math.random().toString(36).substr(2, 9);
  }
}

File: with-kafka/services/fraud-detector/Transaction.py
from dataclasses import dataclass


@dataclass
class Transaction:
    transactionId: str
    senderAccountId: str
    receiverAccountId: str
    amount: int
    currency: str
    timestamp: str

    @staticmethod
    def from_dict(data: dict) -> "Transaction":
        return Transaction(**data)
